### PR TITLE
Timed out operations are removed from funnel

### PR DESCRIPTION
**Pull request by mrooz**

### CODE REVIEW

File: funnel.go [hunk 1]
```diff
@@ -38,6 +42,12 @@ type operationInProcess struct {
 
 	// The result from executing of operation, will be available after the channel will be closed.
 	opResult
+
+	// true when this operation has been deleted from the funnel
+	deleted abool.AtomicBool
```
Comment (ollana) [inline funnel.go]: according to "abool" documentation, we should use pointer to avoid copy

File: funnel.go [hunk 2]
```diff
@@ -129,27 +143,28 @@ func (f *Funnel) getOperationInProcess(operationId string, opExeFunc func() (int
...
 // Closes the operation by updates the operation's result and closure of done channel.
-func (f *Funnel) closeOperation(operationId string) {
+func (f *Funnel) closeOperation(op  *operationInProcess) {
 	f.Lock()
 	defer f.Unlock()
 
-	op, found := f.opInProcess[operationId]
-	if !found {
-		panic("unexpected behavior, operation id not found")
+	//Check if the operation completed after a timeout which would result in the operation being deleted from the funnelhas already deleted the the operation from the funnel.
```
Comment (ollana) [inline funnel.go]: fix comment

File: funnel_test.go [hunk 3]
```diff
@@ -165,3 +166,92 @@ func TestWithTimeout(t *testing.T) {
...
+				return operationId + "ended successfully", errors.New("no error")
+			})
+		}()
+		time.Sleep(requestDelay)
+	}
+
+	wg.Wait()
+
+	elapsedTimeAllRequests :=time.Since(start)
+	expectedOperationTimeoutWithGrace:= funnelTimeout + time.Duration(100*time.Millisecond)
```
Comment (ollana) [inline funnel_test.go:252]: expectedOperationTimeoutWithGrace would be 600 milliseconds, when operationSleepTime is 550. The test doesn't insure the operation ended with a timeout. 
Im suggesting to validate that timeout error returned for each request.

File: funnel.go (hunk 1 above)
Comment (mrooz) [reply funnel.go]: good idea, thanks.

File: funnel.go (hunk 2 above)
Comment (mrooz) [reply funnel.go]: Done.

File: funnel_test.go (hunk 3 above)
Comment (mrooz) [reply funnel_test.go:252]: I'm assuming funnel timeout is working.  Perhaps we need additional tests to assure this.

**Review by ollana — APPROVED** [review]

**Pull request merged**